teos/src/test_utils.rs

```diff
@@ -390,17 +390,25 @@ pub(crate) fn create_carrier(query: MockedServerQuery, height: u32) -> (Carrier,
     )
 }
 
-pub(crate) fn create_responder(
-    tip: ValidatedBlockHeader,
+pub(crate) async fn create_responder(
+    chain: &mut Blockchain,
     gatekeeper: Arc<Gatekeeper>,
     dbm: Arc<Mutex<DBM>>,
     server_url: &str,
 ) -> Responder {
+    let height = if chain.tip().height < IRREVOCABLY_RESOLVED as u32 {
+        chain.tip().height * 2
```

Comment:
If the chain height is less than `IRREVOCABLY_RESOLVED` our cache size will be doubled? I am not getting what this is for!

Reply:
I don't really know what was going on there.

The proper logic here is that the index size needs to be, at least, `IRREVOCABLY_RESOLVED` blocks long. This is due to the assumption that, if a transaction is not found in our cache and not in the mempool, it was confirmed at least `IRREVOCABLY_RESOLVED` blocks ago so we don't need to care about it.

Will update accordingly.